variable.c

```diff
@@ -2672,6 +2672,8 @@ autoload_try_load(VALUE _arguments)
         result = Qfalse;
 
         rb_const_remove(arguments->module, arguments->name);
+
+        rb_raise(rb_eNameError, "autoload failed to resolve %"PRIsVALUE"", QUOTE_ID(arguments->name));
```

Comment:
Currently, an autoload in a module/class shows the outer name when failed.
```console
uninitialized constant X::A (NameError)
```

But this PR loses the information.
Also, `NameError#receiver` and `NameError#name`.

```suggestion
        VALUE module = arguments->module;
        ID name = arguments->name;

        result = Qfalse;

        rb_const_remove(module, name);

        VALUE message = rb_str_new_cstr("autoload failed to resolve ");
        if (module != rb_cObject) rb_str_catf(message, "%"PRIsVALUE"::", module);
        rb_str_append(message, QUOTE_ID(name));
        rb_exc_raise(rb_name_err_new(message, module, ID2SYM(name)));
```